.github/workflows/ci-workflow.yml

```diff
@@ -44,11 +40,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install -e .[dev]
-    - name: Lint with flake8
-      run: |
-        flake8 csp_billing_adapter
-        flake8 tests
-    - name: Test with pytest
+        pip install tox tox-gh-actions
+    - name: Test with tox
       run: |
-        pytest --cov=csp_billing_adapter
+        tox
```

Comment:
A disadvantage that I've seen when combining all checks into one generic named test run is that it's hard to debug failures. If there's a number of different checks happening and one fails you have to scroll the entire output to figure out what the problem is. Having this separated makes that task easier.

The checks for this repo are still small but for comparison https://github.com/SUSE-Enceladus/csp-billing-adapter/actions/runs/5510128319/jobs/10043863281 and https://github.com/SUSE-Enceladus/csp-billing-adapter/actions/runs/5523184440/jobs/10073755309.

Reply:
My suggestion would be to keep the linting and testing separate.

Reply:
Separate as in drive them separately? That is effectively what is happening in most cases...

The Python 3.6 run, thanks to the tox-gh-actions plugin, runs both pytest and flake8 in the same tox run, but in all the other cases, tox will only be running pytest.

If you would prefer them to be explicit, that can be done easily enough.

Reply:
I can change things to have separate `tox -e check` and `tox -e pytest` with explicit which eliminates the need to use tox-gh-actions to control which envs are run for each Python version.

Reply:
Have updated the GitHub Action to explicitly run the specific test envs to run the same tests as before.